agent/config/parse_windows.go

```diff
@@ -32,8 +34,108 @@ const (
 	// domain join check validation. This is useful for integration and
 	// functional-tests but should not be set for any non-test use-case.
 	envSkipDomainJoinCheck = "ZZZ_SKIP_DOMAIN_JOIN_CHECK_NOT_SUPPORTED_IN_PRODUCTION"
+
+	ReleaseId2004SAC        = "2004"
+	ReleaseId1909SAC        = "1909"
+	WindowsServer2019       = "Windows Server 2019"
+	WindowsServer2016       = "Windows Server 2016"
+	WindowsServerDataCenter = "Windows Server Datacenter"
+	InstallationTypeCore    = "Server Core"
+	InstallationTypeFull    = "Server"
+	UnknownOSType           = "unknown"
+	osTypeFormat            = "WINDOWS_SERVER_%s_%s"
+	ecsWinRegistryRootKey   = registry.LOCAL_MACHINE
+	ecsWinRegistryRootPath  = `SOFTWARE\Microsoft\Windows NT\CurrentVersion`
 )
 
+var winRegistry dependencies.WindowsRegistry
+
+func init() {
+	winRegistry = dependencies.StdRegistry{}
+}
+
+func setWinRegistry(mockRegistry dependencies.WindowsRegistry) {
+	winRegistry = mockRegistry
+}
+
+func getInstallationType(installationType string) (string, error) {
+	if InstallationTypeFull == installationType {
+		return "FULL", nil
+	} else if InstallationTypeCore == installationType {
+		return "CORE", nil
+	} else {
+		err := seelog.Errorf("Unsupported Installation type:%s", installationType)
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#error-strings error message should not be capitalized. since you use the one returned from seelog.Errorf, you will need to make sure the log message start with lower case. same for line 78 and line 88 in this file

Reply:
Sure thing. Thank you!